apps/doji-backend/src/auth/auth.service.ts

```diff
@@ -99,4 +66,57 @@ export class AuthService {
       return null
     }
   }
+
+  async sendResetPasswordEmail(email: string): Promise<void> {
+    try {
+      // Find user by email if exists
+      const user = await this.userRepo.findOneOrFail({ email: email })
+      const token = await generateRandomUserToken()
+
+      // Store reset password token in redis
+      const redisKey = generateRedisKey(
+        RedisKeyType.RESET_TOKEN,
+        serializeUserReference({ username: user.username }),
+      )
+      await this.redis.set(redisKey, token)
+      await this.redis.expire(redisKey, RESET_TOKEN_EXPIRE_DURATION_SECONDS)
+
+      // Read reset email template file
+      const resetPasswordTemplate = fs.readFileSync(
+        path.resolve(__dirname, 'assets/reset-password.hbs'),
+        'utf8',
+      )
+      const template = Handlebars.compile(resetPasswordTemplate)
+
+      // Send the email
+      const response = await this.mailgunService.createEmail(environment.mailgun.domain, {
+        from: 'Doji Support <noreply@doji.com>',
+        to: email,
+        subject: '[Doji] Reset Password Request',
+        html: template({ domain: environment.domain, token, username: user.username }),
+      })
+
+      console.log(response)
+    } catch (err) {
+      throw new NotFoundException('User not found')
+    }
```

Comment:
Normally, we just pretend that it was sent anyway as not to acknowledge nor denied the existent of that email in our database. This prevents info leaks.

But I don't give a damn about it.